avoid csp insecure eval on search box.

sbiscigl opened the pull request:
From the issue [issues/6381](https://github.com/doxygen/doxygen/issues/6381) there was a suggestion on an update to the seach.js template to avoid csp violations for `unsafe-eval` which are currently seeing in our published docs. 

`Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self' 'unsafe-inline'".`

This adds the suggested change from the issue, combing through the comments on the issue, i don't see any blocking comments, rather that other places need to be updated unrelated to search. This is to fix the search issue we are seeing. We have created and are using a fork.

albert-github commented on templates/html/search.js:
```diff
@@ -156,7 +156,7 @@ function SearchBox(name, resultsPath, extension)
   this.OnSearchSelectHide = function()
   {
-    this.hideTimeout = setTimeout(this.name +".CloseSelectionWindow()",
+    this.hideTimeout = setTimeout(this.Search.bind(this),
                                   this.closeSelectionTimeout);
```
I have my doubts that this is correct as when on selects the magnifying glass in the search box we get a drop-down box with e.g "All" / "Classes" / "Namespaces" / "Functions" / ... and this drop-down box remains when it looses focus whilst in the current version the drop-down box vanishes.

sbiscigl replied:
yeah you're right, let me investigate a bit

sbiscigl replied:
alright updated the PR and tested for that test case, that drop down box should now disappear when not in focus.

sbiscigl commented:
Hey sorry to bother @albert-github jus wondering if you could take a peek again, i think i addressed the concerns, give a shout if im missing something though

albert-github commented:
I had already a look at it and the result was now OK, though I cannot say / don't know anything about the underlying csp problem

sbiscigl commented:
from testing, at least for us, ti fixed the search/csp issue. apologies as i am a first time contributor here, but curious if you need anything else from me in the review process.

albert-github commented:
Maybe @doxygen has some review comments, I don't have (but like I said I know anything about the underlying csp problem)

doxygen commented:
Looks ok, so I've merged it.

sbiscigl commented:
Thanks!
